### PR TITLE
Jetpack Partner Portal: Introduce payment method page and conditionally add sidebar menu item

### PR DESCRIPTION
**Pull request by rcoll**
**Note:** This PR is a work in progress, and will be broken down into smaller bits before shipping.

#### Changes proposed in this Pull Request

* Introduces the `jetpack/partner-portal-payment` feature flag
* Adds payment method page (feature-flagged) to Jetpack Partner Portal
* Adds sidebar link (feature-flagged) to the sidebar in Jetpack Partner Portal
* Rearranges the sidebar icons per i2 designs

#### Testing instructions

* Load up the partner portal in `jetpack-cloud-development`.
* Ensure that the new sidebar menu item "Payment Method" is present.
* Ensure that when clicking the sidebar menu item you are directed to a new page.
* Ensure you see a list of your account's credit cards (currently inherits WPCOMs stripe account credit cards).
* Ensure you see the "Add credit card form" when visiting that page (You can't actually submit a card yet).
* Observe javascript console for errors and/or warnings.

Related to #

### CODE REVIEW

File: client/jetpack-cloud/sections/partner-portal/primary/payment-method/index.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+		<Main wideLayout className="payment-method">
+			<DocumentHead title={ translate( 'Payment Method' ) } />
+			<SidebarNavigation />
+
+			<div className="payment-method__header">
+				<CardHeading size={ 36 }>{ translate( 'Payment Method' ) }</CardHeading>
+			</div>
+
+			<Card className="payment-method__body">
+				<CardHeading size={ 24 }>{ translate( 'Credit card details' ) }</CardHeading>
```
Comment (a8ci18n) [inline client/jetpack-cloud/sections/partner-portal/primary/payment-method/index.tsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **30** times:
`translate( 'Stored Credit Card Details' )` *ES Score: 9*

**Review by IanRamosC — APPROVED** [review]
LGTM!

File: client/jetpack-cloud/sections/partner-portal/components/credit-card-entry-form/index.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	const [ cardEmail, setCardEmail ] = useState( '' );
+	const [ cardPhone, setCardPhone ] = useState( '' );
+
+	const handleSubmit = ( e ) => {
+		e.preventDefault();
+	};
+
+	return (
+		<div className="credit-card-entry-form">
+			<div className="credit-card-entry-form__header">{ translate( 'Credit card details' ) }</div>
```
Comment (a8ci18n) [inline client/jetpack-cloud/sections/partner-portal/components/credit-card-entry-form/index.tsx:42]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **30** times:
`translate( 'Stored Credit Card Details' )` *ES Score: 9*

File: client/jetpack-cloud/sections/partner-portal/components/credit-card-entry-form/index.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+			<div className="credit-card-entry-form__field">
+				<FormLabel>{ translate( 'Email' ) }</FormLabel>
+				<FormTextInput placeholder="janedoe@gmail.com" />
+			</div>
+			<div className="credit-card-entry-form__field">
+				<FormLabel>{ translate( 'Phone' ) }</FormLabel>
+				<FormTelInput placeholder="(951) 555-0123" />
+			</div>
+			<div className="credit-card-entry-form__field">
+				<FormLabel>{ translate( 'Card details' ) }</FormLabel>
```
Comment (a8ci18n) [inline client/jetpack-cloud/sections/partner-portal/components/credit-card-entry-form/index.tsx:56]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **20** times:
`translate( 'Edit Card Details', { context: 'Header text'} )` *ES Score: 7*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/jetpack-partner-payment)

File: client/jetpack-cloud/sections/partner-portal/components/credit-card-entry-form/index.tsx [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+				<CardElement className="credit-card-entry-form__card-field" />
+			</div>
+
+			<div className="credit-card-entry-form__buttons">
+				<Button onClick={ () => page( '/partner-portal/payment-method' ) }>
+					{ translate( 'Go back' ) }
+				</Button>
+				<Button onClick={ handleSubmit } primary>
+					{ translate( 'Save payment method' ) }
```
Comment (a8ci18n) [inline client/jetpack-cloud/sections/partner-portal/components/credit-card-entry-form/index.tsx:65]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **13** times:
`translate( 'Saved payment method' )` *ES Score: 8*

File: client/jetpack-cloud/sections/partner-portal/primary/payment-method-list/index.tsx [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+					<Gridicon icon="checkmark" size={ 18 } />
+					<div className="payment-method__stored-card-primary">
+						<span className="payment-method__stored-card-primary-text">
+							{ translate( 'Primary' ) }
+						</span>
+					</div>
+					<EllipsisMenu className="payment-method__stored-card-ellipsis" position="bottom">
+						<PopoverMenuItem>{ translate( 'Make primary' ) }</PopoverMenuItem>
+						<PopoverMenuItem>{ translate( 'Update details' ) }</PopoverMenuItem>
+						<PopoverMenuItem>{ translate( 'Delete card' ) }</PopoverMenuItem>
```
Comment (a8ci18n) [inline client/jetpack-cloud/sections/partner-portal/primary/payment-method-list/index.tsx:64]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **23** times:
`translate( 'Delete Card', { context: 'Button label'} )` *ES Score: 10*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/jetpack-partner-payment)

File: client/jetpack-cloud/sections/partner-portal/primary/payment-method-list/index.tsx [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+	const translate = useTranslate();
+
+	return (
+		<div
+			className="payment-method__add-card"
+			onClick={ () => page( '/partner-portal/payment-method/add' ) }
+		>
+			<div className="payment-method__add-card-content">
+				<CardHeading className="payment-method__add-card-text" size={ 18 }>
+					{ translate( 'Add new credit card' ) }
```
Comment (a8ci18n) [inline client/jetpack-cloud/sections/partner-portal/primary/payment-method-list/index.tsx:94]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **23** times:
`translate( 'Add a new credit card' )` *ES Score: 9*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/jetpack-partner-payment)

**Pull request closed without merging**